small-talk/db/registerToDB.js

```diff
@@ -1,28 +1,29 @@
 "use server"
 const uri = "mongodb+srv://smt_root:pokemonwithguns@smalltalkcluster0.jo4jne6.mongodb.net/?retryWrites=true&w=majority"
 import Mongoboi from "./mongo"
-export default async function handleRegister(username, password, firstname, lastname, dob){
+export default async function handleRegister(username, password, firstname, lastname, dob){ // function that sends the credentials to the database
   const mongoboi = new Mongoboi(uri, "Users");
+  const date = new Date(dob).getTime() / 1000; // return the inputted date as a Unix timestamp
   await mongoboi.connect();
 
-  const newUser = {
+  const newUser = { // schema for holding the values into certain fields for database organization
     username: username,
     password: password,
     firstname: firstname,
     lastname: lastname,
-    dob: dob,
+    dob: date,
   };
 
   try {
-    await mongoboi.connect();
+    await mongoboi.connect(); // establish connection to the database
     await mongoboi.insertOne("patients", newUser); // Inserting a new document into the "patients" collection
-    await mongoboi.disconnect();
     return newUser; // Returning the newly inserted user
   } 
-  
   catch (error) {
     console.error("Error writing to database:", error);
-    await mongoboi.disconnect();
     return null; // Returning null if there's an error
   }
+  finally { // ensure that the database connection is closed even if an exception is caught/thrown
```

Comment:
great change, now that the finally block will always run to close the connection.